hibernate-core/src/main/java/org/hibernate/sql/ast/tree/select/SortSpecification.java

```diff
@@ -20,18 +20,15 @@
 	private final Expression sortExpression;
 	private final SortOrder sortOrder;
 	private final NullPrecedence nullPrecedence;
+	private final Collation collation;
```

Comment:
You seem to think that collations only matter for sorting but that's not true, the matter for everything that is based on relational operators and sorting is just one aspect. The DISTINCT operator, relational predicates etc. all depend on the collation. Usually, databases are smart enough to infer a proper collation, but sometimes not, which is when the collation feature as it was before came in.

It's great that you now added tests for this, but I would like to revert the implementation and grammar change.

Reply:
I don't think that, and that's more like the _opposite_ of what this change does.

You're looking at one line of code in isolation and don't realize the context of the code you're looking at.